sdks/java/io/splunk/src/main/java/org/apache/beam/sdk/io/splunk/SplunkIO.java

```diff
@@ -77,32 +78,48 @@
  *
  * PCollection<SplunkWriteError> errors =
  *         events.apply("WriteToSplunk",
- *              SplunkIO.writeBuilder()
- *                     .withToken(token)
- *                     .withUrl(url)
- *                     .withBatchCount(batchCount)
- *                     .withParallelism(parallelism)
- *                     .withDisableCertificateValidation(true)
- *                     .build());
+ *              SplunkIO.write(url, token)
+ *                  .withBatchCount(batchCount)
+ *                  .withParallelism(parallelism)
+ *                  .withDisableCertificateValidation(true));
  * }</pre>
  */
 @Experimental(Kind.SOURCE_SINK)
 public class SplunkIO {
 
+  /**
+   * Write to Splunk's Http Event Collector (HEC).
+   *
+   * @param url splunk hec url
+   * @param token splunk hec authentication token
```

Comment:
A token is required in order to use HEC and each end point has at-least one token. So this would be a required param. (See: [Using http event collector](https://docs.splunk.com/Documentation/Splunk/8.0.4/Data/UsetheHTTPEventCollector#Create_an_Event_Collector_token))